From 24d0ab8a0844922349fa87c9a82684fd5635748a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alfonso=20S=C3=A1nchez-Beato?=
 <alfonso.sanchez-beato@canonical.com>
Date: Thu, 20 Jun 2019 12:09:51 +0200
Subject: [PATCH 16/17] Ensure we do not get duplicated traces on journal

snapd redirectes all process output to the journal, while MM writes its
ouput to both syslog and to stderr. This provokes duplicated traces that
makes debugging harder than needed. We remove the openlog LOG_PERROR so
only the traces written directly by MM appear. Fixes LP: #1833524.
---
 src/mm-log.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mm-log.c b/src/mm-log.c
index 83c5823..71de045 100644
--- a/src/mm-log.c
+++ b/src/mm-log.c
@@ -308,7 +308,7 @@ mm_log_setup (const char *level,
     } else
 #endif
     if (log_file == NULL) {
-        openlog (G_LOG_DOMAIN, LOG_CONS | LOG_PID | LOG_PERROR, LOG_DAEMON);
+        openlog (G_LOG_DOMAIN, LOG_CONS | LOG_PID, LOG_DAEMON);
         log_backend = log_backend_syslog;
     } else {
         logfd = open (log_file,
-- 
2.17.1

